portal-impl/src/com/liferay/portal/util/PortalInstances.java

```diff
@@ -53,6 +55,25 @@
  */
 public class PortalInstances {
 
+	public static Company addCompany(
+			UnsafeSupplier<Company, PortalException> addCompanyUnsafeSupplier,
+			String siteInitializerKey)
+		throws PortalException {
+
+		SiteInitializerThreadLocal.setKey(siteInitializerKey);
```

Comment:
@nNilton what's the scope of this thread local? Although it will be auto reset at the end of the request serving, for the remaining part of the request processing after this method return, the thread local is considered leaked. Please be strict on the scope, and clean it up at the end of it.